felix.go

```diff
@@ -186,6 +186,19 @@ configRetry:
 			time.Sleep(1 * time.Second)
 			continue configRetry
 		}
+
+		// We now have some config flags that affect how we configure the syncer.
+		// After loading the config from the datastore, reconnect, possibly with new
+		// config.  We don't need to re-load the configuration _again_ because the
+		// calculation graph will spot if the config has changed since we were initialised.
+		datastoreConfig = configParams.DatastoreConfig()
```

Comment:
Ah right, I wasn't sure whether there was any config snapshot processing.  (I looked a bit, but not long enough to be sure either way.)

I thought that the `loadConfigFromDatastore` above might be there _because_ of the lack of any config snapshot processing; but I've just realized that it's needed for a different reason: because we need to merge datastore config before entering the config-change-makes-Felix-restart loop, or else Felix would just restart forever.